Implement ColorPicker component

r1g023 opened the pull request.

r1g023 committed:
squashed - rename ColorPicker circle to end with 2

r1g023 commented:
@Valeriusdev @mnichols08 

No need to merge the pull request. I was just testing it to ensure it works on my machine. The next step is to actually implement the color picker circle, but I have a few questions before I start working on it.

KseniiaRiabova commented:
> @Valeriusdev @mnichols08
> 
> No need to merge the pull request. I was just testing it to ensure it works on my machine. The next step is to actually implement the color picker circle, but I have a few questions before I start working on it.

I WILL MERGE, no problem

r1g023 commented:
@Valeriusdev @KseniiaRiabova 

I made sure to pull the latest changes from the remote 'Development' branch before starting work on my own branch to build on the most recent updates. Following your previous feedback, I will avoid creating a PR until my work is complete to prevent confusion. I apologize for rushing earlier; I wanted to show some progress before the meeting. 

Additionally, I squashed my commits into a single commit to remove the initial commit, ensuring there's only one commit reflecting my changes on my branch. I had to force-push this squashed commit to update it remotely on my branch. Please let me know if you have any questions or if anything needs adjustment.

When I pulled from the development branch, I noticed that the footer and other elements were missing. I'm not sure if this was an intentional change by the team or if I missed something over the weekend. To avoid any issues, I haven't modified anything else and have focused solely on implementing the ColorPicker component only and adding that to the App component.

KseniiaRiabova commented:
PR has been closed

r1g023 commented:
@Valeriusdev

As we discussed, I have renamed the Color Circle component to ColorPicker2 because there's already a merged version on the development branch. I also resolved any merge conflicts before pushing to my remote branch. Last but not least, I squashed my previous commits to maintain a clean history. Please review and either approve, close or provide feedback.

Thank you.

Copying @KseniiaRiabova @mnichols08

mnichols08 commented on src/components/ColorPicker2.jsx:
```diff
@@ -0,0 +1,81 @@
+import React, { useRef, useEffect, useState } from "react";
```
we can skip the React import here and just do the modules `{ useRef, useEffect, useState }`

mnichols08 commented on src/components/ColorPicker2.jsx:
Are we planning to use this file? I do not see the component being called within App.jsx.

I am unsure if this is a better approach than the one we have implemented already but we do have this functionality sorted out.

r1g023 replied:
Thank you, @mnichols08, for the feedback. I think others have already completed the task I was assigned, and it has been implemented and integrated into the Dev branch. Regarding this branch, I believe it's okay to close it. My intention was simply to demonstrate activity and gather feedback, which has been helpful. I appreciate you taking the time to review this pull request. I've been traveling over the weekends, so I wasn't able to address this task promptly. It's completely understandable. If possible, I would appreciate it if you could close it.

mnichols08 replied:
I will close it then,  thank you for the update =)

mnichols08 reviewed (commented):
This is a strange PR as we are adding duplicate code. I understand that you wanted to contribute to this task though so I don't want to block it. 

I would like to figure out how to make it a useful PR though. 

Would this approach be better than our current one? Is there anything from it we could implement in our existing project?
